### DISCUSSION TITLE
Functionality for Disabling Services?

### DISCUSSION THREAD

**Discussion by ModestTG** [aksiksi/compose2nix#38] [General]
I had a quick question. I used this tool to convert a Minecraft docker server to Nix. I'm doing this on my desktop PC. I have servers that I could run this on but none of them are as fast or performant as my desktop PC. I was wondering if there was a way to disable the systemd service so that it doesn't start on reboot. I'd like to be able to start this service only when it's in use and be able to stop it to help save some resources if possible. Is something like that able to be done using this tool or would it require a whole re-engineer of how the services are created in systemd?

**Comment by aksiksi** [comment]
Yes, by default, the generated systemd services start on boot. You can disable this re-generating your config with `-auto_start=false`. Let me know if this doesn’t work as I personally don’t test it.

**Reply by ModestTG** [reply]
Ah, yes this should work. I thought I read the options closely to make sure one didn't exist before I asked but I didn't RTFM close enough 😅. Thanks for the quick reply! Great tool!

**Reply by ModestTG** [reply]
I was able to confirm that adding `auto_start=false` seems to both have prevented the service from starting on reboot and on rebuild. Thanks! It is strange though that the service still shows as `enabled`, I guess I would have expected it to not show that? But that could be my lack in understanding of how systemd services work

![image](https://github.com/user-attachments/assets/3a02f075-32b0-410c-a4c2-98bdb767f5f4)

**Reply by aksiksi** [reply]
Yes, normally, you'd disable the systemd service to stop it from starting on boot. But the NixOS module opts to instead empty its `wantedBy` list, which basically means the systemd service is implicitly disabled. Not sure why they do it this way - not a systemd expert haha.

See: https://github.com/NixOS/nixpkgs/blob/nixos-unstable/nixos/modules/virtualisation/oci-containers.nix#L306

**Reply by aksiksi** [reply]
Also, I realized that it would be useful to control auto-start behavior at the Compose service level, so I added a small feature for this.

See: https://github.com/aksiksi/compose2nix?tab=readme-ov-file#auto-start-services-on-boot

**Reply by ModestTG** [reply]
What an awesome feature! Thank you very much!

**Comment by aksiksi** [comment]
Resolved
